base/set.jl

```diff
@@ -191,6 +191,65 @@ function unique(f::Callable, C)
 end
 
 """
+    unique!(A::AbstractVector)
+
+Remove duplicate items as determined by [`isequal`](@ref), then return the modified `A`.
+
+```jldoctest
+julia> unique!([1, 1, 1])
+1-element Array{Int64,1}:
+ 1
+
+julia> A = [7, 3, 2, 3, 7, 5];
+
+julia> unique!(A);
+
+julia> A
+4-element Array{Int64,1}:
+ 7
+ 3
+ 2
+ 5
+```
+"""
+function unique!(A::AbstractVector)
+    if isempty(A)
+        return A
+    elseif issorted(A) || issorted(A, rev=true)
+        # If A is sorted, then we only need to keep track of one element and add that to A
+        # every time that we see a new element.
+        idxs = eachindex(A)
+        m = start(idxs)
+        for i in idxs
+            x = A[i]
+            if x != A[m]
+                _, m = next(idxs, m)
```

Comment:
Right, sorry that was my bad. I have sort of fixed this but there is one point where I have to use `j += 1`, so that will only work if the indices are in ascending order.